lib/pbench/test/functional/agent/cli/commands/conftest.py

```diff
@@ -22,9 +22,10 @@ def pbench_cfg(tmp_path, opt_pbench):
 
 
 @pytest.fixture
-def pbench_run(tmp_path):
+def pbench_run(monkeypatch, tmp_path):
     pbench_agent_d = tmp_path / "var" / "lib" / "pbench-agent"
     pbench_agent_d.mkdir(parents=True, exist_ok=True)
+    monkeypatch.setenv("pbench_run", str(pbench_agent_d))
```

Comment:
This change may have wider effects than we really intend.  If the "normal" operation of our code is to pull the `pbench_run` value from the configuration file, then this change will prevent the unit tests from ever testing that configuration, because it will cause the code to almost always get the value from the environment variable.  Now, perhaps that doesn't matter (because your change will ensure that we test it at least once in each configuration), but my instinct is that the default testing configuration should match our expectation for the default operational configuration, and so we should be setting up the `pbench_run` environment variable only in the few tests which want to check that it works.

Thoughts anyone?

Reply:
That's a good point; I like the idea of a fixture to set this up, but maybe these should be layered, with the original `pbench_run` fixture to create the directory and a layered fixture to use the value of that fixture and establish the environment variable.

The new fixture would be used in the few targeted test cases that need it, while the remainder would continue to operate as they did before.

(And, for what it's worth, as constructed the `yield` is pretty silly; might as well be a `return` unless we're cleaning something up *after* the test ... which I don't believe is necessary since I think the scope of `tmp_path` is per-test.)

Reply:
Of the various open issues, this is the only one that really bothers me; we've implicitly changed the behavior of all the existing tests using `pbench_run`. I'd much prefer to see a new fixture layered on top of `pbench_run` that adds the environment variable, leaving the old behavior unchanged and using this new fixture only for the new test cases specifically intended to check the environment variable behavior.